altair/utils/data.py

```diff
@@ -1,37 +1,61 @@
+from functools import partial
 import json
-import os
 import random
 import hashlib
-import warnings
-from typing import Union, MutableMapping, Optional, Dict, Sequence, TYPE_CHECKING, List
+import sys
+from pathlib import Path
+from typing import (
+    Union,
+    MutableMapping,
+    Optional,
+    Dict,
+    Sequence,
+    TYPE_CHECKING,
+    List,
+    TypeVar,
+    Protocol,
+    TypedDict,
+    Literal,
+    overload,
+    runtime_checkable,
+    Any,
+)
 
 import pandas as pd
-from toolz import curried
-from typing import TypeVar
 
-from ._importers import import_pyarrow_interchange
+from ._importers import import_pyarrow_interchange, import_toolz_function
 from .core import sanitize_dataframe, sanitize_arrow_table, DataFrameLike
 from .core import sanitize_geo_interface
-from .deprecation import AltairDeprecationWarning
 from .plugin_registry import PluginRegistry
 
-
-from typing import Protocol, TypedDict, Literal
-
+if sys.version_info >= (3, 13):
+    from typing import TypeIs
+else:
+    from typing_extensions import TypeIs
 
 if TYPE_CHECKING:
     import pyarrow.lib
 
 
+@runtime_checkable
 class SupportsGeoInterface(Protocol):
     __geo_interface__: MutableMapping
 
 
 DataType = Union[dict, pd.DataFrame, SupportsGeoInterface, DataFrameLike]
 TDataType = TypeVar("TDataType", bound=DataType)
+NonGeoDataType = Union[dict, pd.DataFrame, DataFrameLike]
+NonLikeDataType = Union[dict, pd.DataFrame, SupportsGeoInterface]
```

Comment:
Could we use a more descriptive name than `NonLikeDataType`? Maybe `GeoDataType` if this is only used for geo data (I see that is accepts a regular dataframe too, but in the vegafusion method where it is used, it seems like it only replaces the `__geo_interface__` case)?

Reply:
> Could we use a more descriptive name than `NonLikeDataType`? Maybe `GeoDataType` if this is only used for geo data (I see that is accepts a regular dataframe too, but in the vegafusion method where it is used, it seems like it only replaces the `__geo_interface__` case)?

Absolutely agree it needs a better name!

I was confused seeing `NonLikeDataType` there, but in [this context](https://github.com/dangotbanned/altair/blob/60b8231c4ff785c1fd61918097caff58f8be2cc2/altair/utils/_vegafusion_data.py#L63) it made slightly more sense as its distiguishing two `@overload` defs, that have different return types:

```py
class _ToVegaFusionReturnUrlDict(TypedDict):
    url: str


_VegaFusionReturnType = Union[_ToVegaFusionReturnUrlDict, ToValuesReturnType]


@overload
def vegafusion_data_transformer(
    data: Literal[None] = ..., max_rows: int = ...
) -> Callable[..., Any]: ...


@overload
def vegafusion_data_transformer(
    data: DataFrameLike, max_rows: int # <-------------------------
) -> ToValuesReturnType: ...


@overload
def vegafusion_data_transformer(
    data: NonLikeDataType, max_rows: int # <-------------------------
) -> _VegaFusionReturnType: ...


def vegafusion_data_transformer(
    data: Optional[DataType] = None, max_rows: int = 100000
) -> Union[Callable[..., Any], _VegaFusionReturnType]:
    """VegaFusion Data Transformer"""
    if data is None:
        return vegafusion_data_transformer
    elif isinstance(data, DataFrameLike):
        table_name = f"table_{uuid.uuid4()}".replace("-", "_")
        extracted_inline_tables[table_name] = data
        return {"url": VEGAFUSION_PREFIX + table_name}
    else:
        # Use default transformer for geo interface objects
        # # (e.g. a geopandas GeoDataFrame)
        # Or if we don't recognize data type
        return default_data_transformer(data)
```

Reply:
Fixed in: b0d0e5c6607339db655ef1c5387910adce6a2c46

Reply:
Great, thank you! I agree that it is clearer to write out the three possibilities instead of grouping them under `NonLikeDataType` as in your update..